webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/Score.scala

```diff
@@ -0,0 +1,251 @@
+package com.normation.rudder.score
+
+import com.normation.errors.IOResult
+import com.normation.inventory.domain.Inventory
+import com.normation.inventory.domain.InventoryError.Inconsistency
+import com.normation.inventory.domain.NodeId
+import com.normation.inventory.domain.SoftwareUpdateKind
+import com.normation.rudder.campaigns.CampaignLogger
+import com.normation.rudder.domain.logger.ReportLoggerPure
+import com.normation.rudder.domain.reports.CompliancePercent
+import com.normation.rudder.domain.reports.ComplianceSerializable
+import com.normation.rudder.score.ScoreValue.A
+import com.normation.rudder.score.ScoreValue.B
+import com.normation.rudder.score.ScoreValue.C
+import com.normation.rudder.score.ScoreValue.D
+import com.normation.rudder.score.ScoreValue.E
+import com.normation.zio._
+import zio.Ref
+import zio.ZIO
+import zio.json.ast.Json
+import zio.syntax._
+
+sealed trait ScoreValue {
+  def value: String
+}
+
+object ScoreValue {
+  case object A extends ScoreValue { val value = "A" }
+  case object B extends ScoreValue { val value = "B" }
+  case object C extends ScoreValue { val value = "C" }
+  case object D extends ScoreValue { val value = "D" }
+  case object E extends ScoreValue { val value = "E" }
+
+  val allValues: Set[ScoreValue] = ca.mrvisser.sealerate.values
+
+  def fromString(s: String) = allValues.find(_.value == s.toUpperCase()) match {
+    case None    => Left(s"${s} is not valid status value, accepted values are ${allValues.map(_.value).mkString(", ")}")
+    case Some(v) => Right(v)
+  }
+}
+
+trait Score[T] {
```

Comment:
I don't think we want to have that parametrization. We never use a generic access to details, and `List[Score[_]]` is a signal that we maybe are not doing things like we should. 

Plus, that parametrization risk to create some unpleasant situation since `Score[ComplianceSerializable]` is not the same as `ComplianceScore` (at type level), but in our code, we really really want to have that equality.

Reply:
Something like the first commit in that PR: https://github.com/VinceMacBuche/rudder/pull/29

Reply:
In fact we could remove all ComplianceScore, and juste avec a Score that has a Json as details. Just everything translated into a jsonScore directly from the score event

Reply:
I like the idea to have different score implementation for the different cases, it makes the score computation very clean and really feels like each module can do whatever they want from a structured data type, and not Json. But perhaps I misunderstood ?

Reply:
yes ScoreEvent will be different witth diffrent structure.